src/mon/HealthMonitor.cc

```diff
@@ -161,11 +196,154 @@ bool HealthMonitor::prepare_update(MonOpRequestRef op)
   switch (m->get_type()) {
   case MSG_MON_HEALTH_CHECKS:
     return prepare_health_checks(op);
+  case MSG_MON_COMMAND:
+    return prepare_command(op);
   default:
     return false;
   }
 }
 
+bool HealthMonitor::preprocess_command(MonOpRequestRef op)
+{
+  MMonCommand *m = static_cast<MMonCommand*>(op->get_req());
+  std::stringstream ss;
+  bufferlist rdata;
+
+  cmdmap_t cmdmap;
+  if (!cmdmap_from_json(m->cmd, &cmdmap, ss)) {
+    string rs = ss.str();
+    mon->reply_command(op, -EINVAL, rs, rdata, get_last_committed());
+    return true;
+  }
+
+  MonSession *session = op->get_session();
+  if (!session) {
+    mon->reply_command(op, -EACCES, "access denied", rdata,
+		       get_last_committed());
+    return true;
+  }
+
+  string format;
+  cmd_getval(g_ceph_context, cmdmap, "format", format);
+  boost::scoped_ptr<Formatter> f(Formatter::create(format, "json-pretty",
+						   "json-pretty"));
+
+  string prefix;
+  cmd_getval(g_ceph_context, cmdmap, "prefix", prefix);
+  int r = 0;
+
+//} else {
```

Comment:
???

Reply:
most of the method is boilerplate but there are no read-only commands here yet.  could take it all out i guess but didn't realize at the time we wouldn't implement any

Reply:
actually, i should probably most the "health" implementation from Monitor.cc to here